apps/ui/src/components/Ui/BadgeNetwork.vue

```diff
@@ -5,18 +5,17 @@ import { NetworkID } from '@/types';
 
 const props = withDefaults(
   defineProps<{
-    id: string;
+    id?: string;
     size?: number;
   }>(),
   {
     size: 16
   }
 );
 
-const [network] = props.id.split(':');
-
 const currentNetwork = computed(() => {
   try {
+    const [network] = props.id?.split(':') ?? [];
```

Comment:
Let's just check if `props.id` is defined and otherwise return null. This way we reduce surface where we embrace our broken types 😆